File: LeetCode/144-Binary-Tree-Preorder-Traversal(Medium)/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if (root == null)
            return res;
        
        Stack<TreeNode> stack = new Stack<>();
        stack.push(root);
        
        while (!stack.isEmpty()) {
            TreeNode node = stack.pop();
            res.add(node.val);
            
            if (node.right != null) stack.push(node.right);
            if (node.left != null) stack.push(node.left);
        }
        
        return res;
    }
}

File: LeetCode/144-Binary-Tree-Preorder-Traversal(Medium)/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> res;
        if (root == NULL) return res;
        
        stack<TreeNode*> stack;
        
        stack.push(root);
        
        while (!stack.empty()) {
            TreeNode* curr = stack.top();
            stack.pop();
            
            res.push_back(curr->val);
            
            if (curr->right != NULL)stack.push(curr->right);
            if (curr->left != NULL)stack.push(curr->left);
        }
        
        return res;
    }
};